src/profile-logic/profile-data.js

```diff
@@ -569,6 +569,234 @@ export function filterThreadToPrefixCallNodePath(
 }
 
 /**
+ * Transform a thread's stacks to merge stacks that match the `prefixCallNodePath` into
+ * the calling stack. See `src/types/transforms.js` for more information about the
+ * "merge-call-node" transform.
+ */
+export function mergeCallNode(
+  thread: Thread,
+  prefixCallNodePath: IndexIntoFuncTable[],
+  implementation: ImplementationFilter
+): Thread {
+  return timeCode('mergeCallNode', () => {
+    const { stackTable, frameTable, samples } = thread;
+    // Depth here is 0 indexed.
+    const depthAtCallNodePathLeaf = prefixCallNodePath.length - 1;
+    const oldStackToNewStack: Map<
+      IndexIntoStackTable | null,
+      IndexIntoStackTable | null
+    > = new Map();
+    oldStackToNewStack.set(null, null);
+    const newStackTable = {
+      length: 0,
+      prefix: [],
+      frame: [],
+    };
+    const stackDepths = [];
+    const stackMatches = [];
```

Comment:
Q: are you using 2 arrays of simple values instead of 1 array of objects so that it's easier to the GC ?

As I understand they're just caches for the algorithm, could you add a comment for this ?